contacts-app/src/components/ContactForm/ContactForm.jsx

```diff
@@ -14,65 +16,116 @@ class ContactForm extends React.Component {
         companyName: "",
         phone: "",
         email: ""
+      },
+      invalidMessage: {
+        firstName: "",
+        lastName: "",
+        phone: "",
+        email: ""
       }
     };
   }
 
   handleInputChange = event => {
     const update = {};
-    update[event.target.id] = event.target.value;
+    const inputId = event.target.id;
+    const inputValue = event.target.value;
+    if (inputId === "First Name" || "Last Name") {
+      update[inputId] =
+        inputValue.charAt(0).toUpperCase() + inputValue.slice(1);
+    } else {
+      update[inputId] = inputValue;
+    }
     this.setState({
       newContact: { ...this.state.newContact, ...update }
     });
   };
 
   handleAddContact = event => {
     event.preventDefault(); // to prevent refresh the page
-    this.props.addNewContact({
-      ...this.state.newContact,
-      id: this.props.contactId
-    });
+    const isContactFormInvalid = this.validateContactForm();
+    !isContactFormInvalid &&
+      this.props.addNewContact({
+        ...this.state.newContact,
+        id: this.props.contactId
+      });
   };
+
+  validateContactForm() {
+    const { firstName, lastName, phone, email } = this.state.newContact;
+    //regex for validate email with @, domain and no space.
+    const isEmailInvalid = /^\S+@\S+\.\S+$/.exec(email) === null;
+    const isFirstNameInvalid = firstName.trim() === "";
+    const isLastNameInvalid = lastName.trim() === "";
+    //regex for validate email xxx-xxx-xxxx  or xxx xxx xxxx format
+    const phoneRegex = /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/;
+    const isPhoneInvalid = !phoneRegex.test(phone);
+    this.setState({
+      invalidMessage: {
+        firstName: isFirstNameInvalid ? "First name cannot be empty" : "",
+        lastName: isLastNameInvalid ? "Last name cannot be empty" : "",
+        phone: isPhoneInvalid
+          ? "Invalid phone number, please enter number in xxx-xxx-xxxx or xxx xxx xxxx format"
+          : "",
+        email: isEmailInvalid ? "Invalid Email" : ""
+      }
+    });
+    if (
+      isFirstNameInvalid &&
+      isLastNameInvalid &&
+      isPhoneInvalid &&
+      isEmailInvalid
+    )
+      return true;
+    if (isFirstNameInvalid) return true;
+    if (isLastNameInvalid) return true;
+    if (isPhoneInvalid) return true;
+    if (isEmailInvalid) return true;
```

Comment:
You can just do OR's here

```js
return (
      isFirstNameInvalid ||
      isLastNameInvalid ||
      isPhoneInvalid ||
      isEmailInvalid
    );
```

Reply:
always find `||` & `&& `can be mind twisting